### PR TITLE
Add a script to check for dead code

### PR DESCRIPTION
**Pull request by SimonBrandner**
Towards https://github.com/vector-im/element-web/issues/21969

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,8 @@
     "lint:types": "tsc --noEmit --jsx react && tsc --noEmit --project ./tsconfig.module_system.json",
     "lint:style": "stylelint \"res/css/**/*.scss\"",
     "test": "jest",
-    "coverage": "yarn test --coverage"
+    "coverage": "yarn test --coverage",
+    "prune:code": "node ./scripts/prune_code.js"
```
Comment (weeman1337) [inline package.json:55]: While the utility we use behind the scenes is `ts-prune` I find the name „prune“ for this script misleading. It does actually not prune code. What about naming it something like „analyse:unused-exports“.

**Review by weeman1337 — COMMENTED** [review]
Looks good :+1:  Some minor things (see my comments).

File: package.json (hunk 1 above)
Comment (SimonBrandner) [reply package.json:55]: I was struggling with the naming a bit :sweat_smile: Your suggestion sounds great - thank you!

**Review by weeman1337 — APPROVED** [review]
LGTM :+1:

**Pull request merged**